frontend/hub/collections/hooks/useCopyToRepository.tsx

```diff
@@ -0,0 +1,252 @@
+  const pulpRequest = useGetRequest<PulpItemsResponse<SigningServiceResponse>>();
+
+  const [selectedRepositories, setSelectedRepositories] = useState<Repository[]>([]);
+  const [fixedRepositories, setFixedRepositories] = useState<Repository[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>('');
+
+  const { collection_auto_sign, require_upload_signatures } = props.context.featureFlags;
+  const autoSign = collection_auto_sign && !require_upload_signatures;
+
+  const operation = props.operation;
+
+  const copyToRepositories = async () => {
+    try {
+      setIsLoading(true);
+
+      const { repository } = props.collection;
+      if (!repository) {
+        return;
+      }
+
+      const pulpId = parsePulpIDFromURL(repository?.pulp_href);
+
+      let signingService = '';
+
+      if (autoSign) {
+        const signingServiceName = props.context.settings.GALAXY_COLLECTION_SIGNING_SERVICE;
+
+        const url = pulpAPI`/signing-services/?name=${signingServiceName}`;
+        const signingServiceList = await pulpRequest(url);
+
+        signingService = signingServiceList?.results?.[0].pulp_href;
+      }
+
+      const repoHrefs: string[] = [];
+
+      for (const repo of selectedRepositories) {
+        if (!fixedRepositories.find((item) => item.name == repo.name)) {
+          repoHrefs.push(repo.pulp_href);
+        }
+      }
+
+      const params: {
+        collection_versions: string[];
+        destination_repositories: string[];
+        signing_service?: string;
+      } = {
+        collection_versions: collection.collection_version?.pulp_href
+          ? [collection.collection_version?.pulp_href]
+          : [],
+        destination_repositories: repoHrefs,
+      };
+
+      if (signingService) {
+        params.signing_service = signingService;
+      }
+
+      let api_op = operation as string;
+      if (operation == 'approve') {
+        api_op = 'move';
+      }
+
+      // TODO test errors
+      await hubAPIPost(
+        pulpAPI`/repositories/ansible/ansible/${pulpId || ''}/${api_op}_collection_version/`,
+        params
+      );
+
+      setIsLoading(false);
+      props.onClose();
+    } catch (error) {
+      setError(
+        operation == 'approve' ? t('Can not approve collection.') : t('Can not copy collection')
+      );
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    async function getSelected() {
+      const repos = await request(
+        hubAPI`/v3/plugin/ansible/search/collection-versions?limit=100000&name=${
+          collection.collection_version?.name || ''
+        } &&version=${collection.collection_version?.version || ''}`
+      );
+
+      if (repos.data?.length > 0) {
+        const reposMapped = repos.data.map((item) => item.repository || ({} as Repository));
+        setFixedRepositories(reposMapped);
+      }
+    }
+
+    void (async () => {
+      await getSelected();
+    });
```

Comment:
```suggestion
    getSelected();
```

(unless this is a bug and was supposed to await?)

Reply:
(possibly the whole thing could be rewrritten to

```js
useEffect(() => {
  const { name, version } = collection!.collection_version;
  request(hubAPI`/v3/plugin/ansible/search/collection-versions?limit=100000&name=${name}&version=${version}`)
    .then(({ data }) => {
      if (data.length) {
        const reposMapped = data.map((item) => item.repository || ({} as Repository));
        setFixedRepositories(reposMapped);
      }
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```

Reply:
Well I must say I like await much more than then, it seems readable to me. I use promises directly only when I need to.